src/controllers/transaction/TransactionMaterialController.ts

```diff
@@ -51,16 +54,43 @@ export default class TransactionMaterialController extends AbstractController<Tr
 	 * @param res Express Response
 	 */
 	private getTransactionMaterial: RequestHandler = async (
-		{ query: { noMeta, at } },
+		{ query: { noMeta, at, metadata } },
 		res
 	): Promise<void> => {
 		const hash = await this.getHashFromAt(at);
 
-		const noMetaArg = noMeta === 'true';
+		const metadataArg = this.parseMetadataArgs(noMeta, metadata);
 
 		TransactionMaterialController.sanitizedSend(
 			res,
-			await this.service.fetchTransactionMaterial(hash, noMetaArg)
+			await this.service.fetchTransactionMaterial(hash, metadataArg)
 		);
 	};
+
+	/**
+	 * The metadata args have two
+	 *
+	 * @param noMeta
+	 * @param metadata
+	 */
+	private parseMetadataArgs(
+		noMeta: unknown,
+		metadata: unknown
+	): MetadataOpts | false {
+		const metadataArg =
+			!metadata || metadata === 'json' || metadata === 'scale';
+
+		if (!metadataArg)
+			throw new Error('Invalid inputted value for the `metadata` query param.');
+		if (metadata === 'json') return 'json';
+		if (!noMeta || noMeta === 'false' || metadata === 'scale') return 'scale';
+		if (noMeta === 'true') {
+			Log.logger.warn(
+				'`noMeta` query param will be deprecated in sidecar v13, and replaced with `metadata` please migrate'
+			);
+			return false;
+		}
+
+		return 'scale';
```

Comment:
I was pondering whether this could be simplified. The best I came up with was:

```
if (!noMeta && metadata === 'json')
    return 'json';
if (!noMeta && metadata === 'scale')
    return 'scale';
if (!noMeta && metadata)
    throw new Error('Invalid inputted value for the `metadata` query param.');

if (noMeta) {
    Log.logger.warn(
        '`noMeta` query param will be deprecated in sidecar v13, and replaced with `metadata` please migrate'
    );
}
return false;
```

Reply:
Yea I agree, when I wrote this I knew it was really late, and super messy looking and was in need of a re-write 😅 . Thank you for working through that,  I had a lot of trouble grouping the arguments together. I agree the above is significantly better and ill update it!

Reply:
So the only issue with the logic above is that we need to account for when no query params are inputted the default behavior is to have scale metadata returned. 

```
/**
		 * Checks to see if the `metadata` query param is inputted, if it isnt,
		 * it will default to the old behavior. This is to be removed once after 
		 * the `noMeta` query param is fully deprecated.
		 */
		if (metadata) {
			switch (metadata) {
				case('json'):
					return 'json';
				case('scale'):
					return 'scale';
				default:
					throw new Error('Invalid inputted value for the `metadata` query param.');
			}
		}

		if (noMeta) {
			Log.logger.warn(
				'`noMeta` query param will be deprecated in sidecar v13, and replaced with `metadata` please migrate'
			);
			if (noMeta === 'true') return false;
			if (noMeta === 'false') return 'scale';
		}

		// default behavior until `noMeta` is deprecated, then false will be default
		return 'scale';
```

Reply:
So i updated the logic a little, and kept the same format :). Thanks for making this easy for me to understand!

Reply:
Also the idea here being that we want the metadata query param to override `noMeta` if its used

Reply:
Aaah overriding noMeta, got it! your new version looks clean :)